prefect_azure/container_instance.py

```diff
@@ -0,0 +1,704 @@
+            "task definition."
+        ),
+    )
+    # Execution settings
+    task_start_timeout_seconds: int = Field(
+        default=240,
+        description=(
+            "The amount of time to watch for the start of the ACI container. "
+            "before marking it as failed."
+        ),
+    )
+    task_watch_poll_interval: float = Field(
+        default=5.0,
+        description=(
+            "The amount of time to wait between Azure API calls while monitoring the "
+            "state of an Azure Container Instances task."
+        ),
+    )
+
+    @validator("command")
+    def validate_command(cls, command: List[str]):
+        """
+        A Pydantic validator that ensures a command is available to run
+        in an Azure container instance.
+
+        Args:
+            command: The command to validate.
+
+        Returns:
+            A command list if present, otherwise the base flow run command.
+        """
+
+        if not command:
+            return cls._base_flow_run_command()
+        return command
+
+    @sync_compatible
+    async def run(
+        self, task_status: Optional[TaskStatus] = None
+    ) -> ContainerInstanceJobResult:
+        """
+        Runs the configured task using an ACI container.
+
+        Args:
+            task_status: An optional `TaskStatus` to update when the container starts.
+
+        Returns:
+            An `ContainerInstanceJobResult` with the container's exit code.
+        """
+
+        run_start_time = datetime.datetime.now(datetime.timezone.utc)
+        token_credential = self._create_credential()
+        aci_client = self._create_container_client(token_credential)
+        container = self._configure_container()
+        container_group = self._configure_container_group(token_credential, container)
+        created_container_group = None
+
+        self.logger.info(
+            f"{self._log_prefix}: Preparing to run command {' '.join(self.command)!r} "
+            f"in container {container.name!r} ({self.image})..."
+        )
+        try:
+            self.logger.info(f"{self._log_prefix}: Waiting for container creation...")
+            # Create the container group and wait for it to start
+            creation_status_poller = aci_client.container_groups.begin_create_or_update(
+                self.resource_group_name, container.name, container_group
+            )
+            created_container_group = await run_sync_in_worker_thread(
+                self._wait_for_task_container_start, creation_status_poller
+            )
+
+            # If creation succeeded, group provisioning state should be 'Succeeded'
+            # and the group should have a single container
+            if self._provisioning_succeeded(created_container_group):
+                self.logger.info(f"{self._log_prefix}: Running command...")
+                if task_status:
+                    task_status.started(value=container.name)
```

Comment:
Is the container name unique? Will we be able to look up the container by this name later to check its status?

Reply:
Yes I think it's unique https://github.com/PrefectHQ/prefect-azure/pull/45/files#diff-d0fb6676743e33e5f40344430bec8c2fc1e24b09d78584f96d7adf450fc27dafR344

Reply:
It is; Azure requires that container names are unique within each resource group, so we're using a UUID as the container name. 

But if we want something we can use to look up the container laser, we probably want container group name instead of container name. It is generated by Azure, and is also unique. The container instances API doesn't let you pull a single container's details directly; instead, you load the container group and get back details for all the containers in the group. 

We're using a unique container group with a single container for each flow run, as this what the Azure docs recommended for our use case, so knowing the container group name would make it easy to look up the flow run container.

Reply:
Ok, sounds good. It is now updated to use container group name.